pytest_bdd/parser.py

```diff
@@ -457,4 +456,5 @@ def get_tags(line):
     return {tag.lstrip("@") for tag in line.strip().split(" @") if len(tag) > 1}
 
 
-STEP_PARAM_RE = re.compile(r"\<(.+?)\>")
+STEP_PARAM_TEMPLATE = "<{param}>"
+STEP_PARAM_RE = re.compile(STEP_PARAM_TEMPLATE.format(param="([^<>]+?)"))
```

Comment:
I wonder what was "?" doing here. Is what between <> even optional?

Reply:
*?, +?, ??
The '*', '+', and '?' qualifiers are all greedy; they match as much text as possible. Sometimes this behaviour isn’t desired; if the RE <.*> is matched against '<a> b <c>', it will match the entire string, and not just '<a>'. Adding ? after the qualifier makes it perform the match in non-greedy or minimal fashion; as few characters as possible will be matched. Using the RE <.*?> will match only '<a>'.

Reply:
For example:
<<nested parameter> partial outer parameter> this regex is matched to "nested parameter". Without ? it would match "nested parameter> partial outer parameter" which is undesired